visualization/app/codeCharta/ui/ribbonBar/ribbonBar.component.ts

```diff
@@ -1,18 +1,41 @@
 import "./ribbonBar.component.scss"
+import { IRootScopeService } from "angular"
 import $ from "jquery"
+import { StoreService } from "../../state/store.service"
+import { PanelSelection } from "../../codeCharta.model"
+import { setPanelSelection } from "../../state/store/appSettings/panelSelection/panelSelection.actions"
+import { PanelSelectionService, PanelSelectionSubscriber } from "../../state/store/appSettings/panelSelection/panelSelection.service"
+import { CodeChartaMouseEventService } from "../../codeCharta.mouseEvent.service"
 
-export class RibbonBarController {
-	private readonly EXPANDED_CLASS = "expanded"
+export class RibbonBarController implements PanelSelectionSubscriber {
+	constructor(
+		private storeService: StoreService,
+		private $rootScope: IRootScopeService,
+		private codeChartaMouseEventService: CodeChartaMouseEventService
+	) {
+		PanelSelectionService.subscribe(this.$rootScope, this)
+	}
 
-	public toggle(event) {
-		const boxElement = $(event.srcElement).closest("md-card")
-		$(document.activeElement).blur()
+	private _viewModel: {
```

Comment:
Using an underscore to indicate private methods is not ideal. Instead, use an actual private property `#viewModel`.

Reply:
I never really thought about why we'd put an underscore before that. Since I'm working with codeCharta, viewModels were always prefixed with an underscore. I think they shouldn't prefixed at all.

Can private properties from es6-classes be used with typescript and set to a type, e.g. `#viewModel: MY_TYPE = {}`?
I don't think our configs support private properties right now and using them would require a huge refactoring in all classes.

Reply:
I agree that they should not be prefixed.

https://www.typescriptlang.org/docs/handbook/classes.html#ecmascript-private-fields

If it's too much work to refactor, let's not do that now :-)